src/Device/Driver/XCVario.cpp

```diff
@@ -158,6 +160,23 @@ XVCDevice::ParseNMEA(const char *String, NMEAInfo &info)
     return false;
 }
 
+// For documentation refer to chapter 10.1.3 Device Driver/XCVario in mulilingual handbook: https://xcvario.de/handbuch
+
+bool 
+XVCDevice::PutQNH(const AtmosphericPressure &pres, OperationEnvironment &env)
+{
+  /* the XCVario understands "!g,q<NNNN>" command for QNH updates with recent builds */
+  char buffer[32];
+  unsigned qnh = uround(pres.GetPascal()/100);
+  int msg_len = sprintf(buffer,"!g,q%u\r", std::min(qnh,(unsigned)2000));
+  if ( msg_len > 0 ) {
+    return port.FullWrite(buffer, msg_len, env, std::chrono::seconds(2) );
+  }
+  return false;
```

Comment:
How is this reachable?

Reply:
The idea behind this was to check formatting errors. I agree, in this trivial case this should not happen for possible range of an integer 0..2000 , the sprintf %u should of course always work. We can of course remove this double check, okay ?

Reply:
What kind of formatting error would make `sprintf()` fail? I ask because maybe you have knowledge about this that I don't have, and then we need to fix a thousand `printf()` calls in XCSoar. If you can't explain how it can fail, then we don't need the check.

Of course, I know how printf can fail - but such failures remain undetected; for example, if the buffer is too small. `sprintf()` will overwrite the buffer bounds, which can lead to all kinds of behavior, but *not* to a negative return value. `snprintf()` on the other hand will silently truncate the output, which also doesn't return a negative value. There are various other kinds of mistakes, like not enough arguments, wrong argument types etc. - but none of these lead to a negative return value, but to other random behavior, including crashes. All of these are not runtime errors, but code bugs, therefore a runtime check is the wrong way to approach this.

Reply:
Great explanation, you had already won, change is addressed in here:
https://github.com/XCSoar/XCSoar/pull/766/commits/b79114fc91f61eabcb363d4a2b44a65b7489bbc1

Reply:
This isn't about winning - when I see code that looks strange, I want to know the author's motivation, to see whether they had something in mind that I was not aware about. I want to learn and self-improve, that's why I ask seemingly stupid questions. When it turns out that the code was just strange and unnecessarily complex with no motivation/reason at all, like here, I havn't "won". Oh yeah, I've been right after all, but who cares. Only you have "won", because it was your chance to learn something, you got your takeaway, but I got nothing.

Reply:
Me and @hjr appreciate the way you help not only us to improve things and understand how to get out of the way of bugs, issues or simply performance limiting unneeded code, by asking the right questions. The next pull request shall consider all the lessons learned!